hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ReplaceCommitValidateUtil.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.action.cluster;
+
+import org.apache.hudi.common.model.HoodieReplaceCommitMetadata;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.JsonUtils;
+import org.apache.hudi.exception.HoodieException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.apache.hudi.common.util.FileIOUtils.LOG;
+
+public class ReplaceCommitValidateUtil {
+  public static final String REPLACE_COMMIT_FILE_IDS = "replaceCommitFileIds";
```

Comment:
I'm thinking we can directly use `replaceCommitMetadata.partitionToReplaceFileIds` given the inflight replace commits actually don't use it. This way we don't need to validate it differently for `inflight` and `complete`.

Reply:
This is because it appears that there is no relevant file ID information in the inflight status. However, if there are two plans being executed simultaneously and not yet submitted, can we prevent this situation from occurring by performing additional validation on the inflight status?

Reply:
How you validate it if such issue happens? Like 2 inflight commits written nearly at the same time, and each other doesn't see it?

Reply:
I don't think this situation will happen:
Because my validation was done after the inflight commit was written to disk. Suppose two inflight replace commits happen at the same time:
1. Simultaneous disk write: Even if the disk write happens at the same time, when validation begins, it will reload first, so it will definitely see the file that the other party has written to disk.
2. Simultaneous validation: This is even if both have written to disk, both validation will find each other and report errors.
3. One party validates while the other writes to disk: This means that the verifying party has already completed the disk write, so the committing party will definitely be found.